node-graph/gstd/Cargo.toml

```diff
@@ -91,3 +91,5 @@ autoquant = { git = "https://github.com/truedoctor/autoquant", optional = true,
 
 # Optional dependencies
 image-compare = { version = "0.4.1", optional = true }
+imageproc = "0.25.0"
+ndarray = "0.15.6"
```

Comment:
Oof sorry about imageproc I forgot to remove it before committing, as for ndarray it is possible to do it manually but I think it is a useful addition because it makes implementing image processing algorithms significantly simpler with the data structures and functions it provides however I can do it manually if necessary.

Reply:
@Keavon thoughts?

Reply:
If it's not a huge hassle, I'd love to see a diff comparing its removal. And to know if there'd be likely any performance difference. If rewriting it without ndarray would be a lot of work that's potentially wasted if we keep the crate, just say so— I don't want you to waste time on it. I can investigate its dependency chain and tradeoffs more later today.

Reply:
Having a simple ndarray is probably quite useful for other nodes so I'm personally happy to add this dependency.

Reply:
@0HyperCube I was thinking potentially along those same lines too, but I will want to get a bit more info when I can dig into it more (and hopefully see a comparison if it'd be easy). Thanks for your thoughts, that's helpful.

Reply:
We could also consider using this crate: https://crates.io/crates/micro_ndarray. This should basically give us the same features at a much smaller code size and with zero deps

Reply:
@Keavon sure thing, I'll do a comparison of the crate vs doing it manually this week.

@TrueDoctor micro_ndarray looks promising as well I'll look into it

Reply:
One other option: if `micro_ndarray` uses the same API (for the parts we use) as `ndarray`, you could just make it a drop-in replacement and push it. Then I'd be willing to just approve that library and save us the time of testing out the library-free version, since the `micro` version is a decent compromise.

Reply:
It's not a drop in replacement unfortunately but if you're willing to approve that library I'll just reimplement the node using it wouldn't be much work to do so

Reply:
How many dimensions are the arrays we're working with here? Just curious because of how it `micro_ndarray` talks about better perf at higher dimensionality compared to `ndarray`.